runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/coords/Rs2WorldPoint.java

```diff
@@ -107,6 +122,97 @@ public int distanceToPath(Client client, WorldPoint other)
         return distance;
     }
 
+    /**
+     * Finds the nearest walkable tile around a specified game object that the player can interact with.
+     *
+     * <p>This method calculates the closest walkable tile adjacent to the given game object, considering the player's current position.
+     * It ensures that both the player and the object are on the same plane before proceeding. The method retrieves interactable points
+     * around the object, filters out non-walkable tiles, and selects the closest one to the player.</p>
+     *
+     * @param tileObject The {@link GameObject} for which to find the nearest walkable tile.
+     * @return An {@link Rs2WorldPoint} representing the nearest walkable tile around the object, or {@code null} if none are found.
+     */
+    public static Rs2WorldPoint getNearestWalkableTile(GameObject tileObject) {
```

Comment:
I was thinking that this should live in Rs2Tile.java

Wdyt?

Reply:
Yeah you're probably right, it does fit in well in Rs2Tile.
My reasoning was to put the method in the same class as the return type. 

I'll move it to Rs2Tile instead :)